overlord/snapstate/backend/link.go

```diff
@@ -47,13 +47,25 @@ func updateCurrentSymlinks(info *snap.Info) error {
 		logger.Noticef("Cannot remove %q: %v", currentDataSymlink, err)
 	}
 
-	if err := os.MkdirAll(info.DataDir(), 0755); err != nil {
+	if err := os.MkdirAll(dataDir, 0755); err != nil {
 		return err
 	}
+	cleanup := []string{dataDir, ""}[:1] // cleanup has cap(2)
+	defer func() {
+		if e == nil {
+			return
+		}
+		for _, d := range cleanup {
+			if err := os.Remove(d); err != nil {
+				logger.Noticef("Cannot clean up %q: %v", d, err)
+			}
+		}
```

Comment:
Should the cleanup restore the 'old' symlinks too?

Reply:
that is done much later in the undo chain if I remember correctly, the old links are removed when we deactived the snap with unlink-current-snap and its undo would put them back or so I think I remember